app/webpacker/components/SearchWidget/WcaSearch.jsx

```diff
@@ -16,21 +12,7 @@ export default function WcaSearch({
   model,
   params,
 }) {
-  const urlFn = useCallback((query) => {
-    if (model === 'user') {
-      return `${userSearchApiUrl(query)}&${new URLSearchParams(params).toString()}`;
-    }
-
-    if (model === 'person') {
-      return `${personSearchApiUrl(query)}&${new URLSearchParams(params).toString()}`;
-    }
-
-    if (model === 'competition') {
-      return `${competitionSearchApiUrl(query)}&${new URLSearchParams(params).toString()}`;
-    }
-
-    throw new Error(`Invalid search type in WcaSearch component: ${model}`);
-  }, [params, model]);
+  const urlFn = useCallback((query) => apiV0Urls.search(query, [model], params), [model, params]);
```

Comment:
Hm... I get that your new version looks cleaner, but it entirely discards the specialized search APIs.
Right now, these APIs do nothing more than forward to the general search method, but in the future we might leverage the advantage of having sspecialized endpoints for specialized caching or whatever.

Please consider adding back the previous version or make an argument in favor of the new version.

Reply:
Okay added back the previous version.